src/mscorlib/shared/System/Numerics/Vector.cs

```diff
@@ -763,6 +765,17 @@ private Vector(ref Register existingRegister)
         {
             this.register = existingRegister;
         }
+
+        /// <summary>
+        /// Constructs a vector from the given span.
+        /// The span must contain at least Vector'T.Count elements.
+        /// </summary>
+        [Intrinsic]
+        public unsafe Vector(Span<T> values)
+            : this()
+        {
+            this = Unsafe.ReadUnaligned<Vector<T>>(ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(values)));
```

Comment:
Do we need to validate that T is valid vector type here, similar how array constructors validate it?

Otherwise, I will be able to do things like:
```
Span<string> s = new string[] { "a", ... };
Vector<string> v = new Vector<string>(s);
```
that is going bad things to happen.

Reply:
The Vector definition is as follows. So will that exclude the above cases...?

```C#
public struct Vector<T> : IEquatable<Vector<T>>, IFormattable where T : struct
```

Reply:
You are right that will exclude my example with string, but it won't exclude other structs, e.g.:

```
Span<Guid> s = 
Vector<Guid> v = new Vector<Guid>(s);
```

Reply:
Thanks. Earlier I missed the importance of `supportedTypes` defined in the GenerationConfig.ttinclude file; which is used in the `T[]` constructor already. I will use the same construct here.